lemur/plugins/lemur_aws/acm.py

```diff
@@ -0,0 +1,178 @@
+            return client.import_certificate(
+                Certificate=str(body),
+                PrivateKey=str(private_key),
+                CertificateChain=str(cert_chain),
+            )
+        else:
+            return client.import_certificate(
+                Certificate=str(body),
+                PrivateKey=str(private_key),
+            )
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] != "EntityAlreadyExists":
+            raise e
+
+
+@sts_client("acm")
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def delete_cert(cert_arn, **kwargs):
+    """
+    Delete a certificate from ACM AWS
+
+    :param cert_arn:
+    :return:
+    """
+    client = kwargs.pop("client")
+    metrics.send("delete_cert", "counter", 1, metric_tags={"cert_arn": cert_arn})
+    try:
+        client.delete_certificate(CertificateArn=cert_arn)
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] != "NoSuchEntity":
+            raise e
+
+
+@sts_client("acm")
+def get_certificate(name, **kwargs):
+    """
+    Retrieves an acm SSL certificate.
+
+    :return:
+    """
+    return _get_certificate(name, **kwargs)
+
+
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def _get_certificate(arn, **kwargs):
+    metrics.send("get_certificate", "counter", 1, metric_tags={"arn": arn})
+    client = kwargs.pop("client")
+    try:
+        return client.get_certificate(CertificateArn=arn)
+    except client.exceptions.NoSuchEntityException:
+        capture_exception()
+        return None
+
+
+@sts_client("acm")
+def get_certificates(**kwargs):
+    """
+    Fetches one page of acm certificate objects for a given account.
+    :param kwargs:
+    :return:
+    """
+    return _get_certificates(**kwargs)
+
+
+@retry(retry_on_exception=retry_throttled, wait_fixed=2000, stop_max_attempt_number=25)
+def _get_certificates(**kwargs):
+    metrics.send("get_certificates", "counter", 1)
+    return kwargs.pop("client").list_certificates(
+        **kwargs,
+        CertificateStatuses = [
+            'ISSUED'
+        ]
+    )
+
+
+@sts_client("acm")
+def get_all_certificates(**kwargs):
+    """
+    Use STS to fetch all of the ACM SSL certificates from a given account
+    :param restrict_path: If provided, only return certificates with a matching Path value.
+    """
+    certificates = []
+    account_number = kwargs.get("account_number")
+    metrics.send(
+        "get_all_certificates",
+        "counter",
+        1,
+        metric_tags={"account_number": account_number},
+    )
+
+    while True:
+        response = _get_certificates(**kwargs)
+        metadata = response["CertificateSummaryList"]
+
+        for m in metadata:
+            certificate = _get_certificate(
+                m["CertificateArn"],
+                client=kwargs["client"]
+            )
+
+            certificate.update(
```

Comment:
in case we return a None due to NoSuchEntityException, we should `continue` instead of throwing an exception calling update on None

Reply:
Done!